lib/vanagon/platform/dsl.rb

```diff
@@ -405,6 +405,20 @@ def zypper_repo(definition)
       def add_build_repository(*args)
         @platform.add_build_repository(*args)
       end
+
+      # Adds an override to a build requirement. When component.build_requires is
+      # called, the component will check to see if the requirement matches one of
+      # these overrides, and use the override if there's a match.
+      #
+      # @param build_requirement [String] build requirement to overrides
+      # @param override [String] value of the override
+      # @raise [Vanagon::Error] an arror is raised if an attempt is made to override a requirement that already has one
+      def override_build_requirement(build_requirement, override)
+        if @platform.build_requirement_overrides.key?(build_requirement)
+          raise Vanagon::Error, "build requirement already overridden by #{@platform.build_requirement_overrides[build_requirement]}"
```

Comment:
not entirely sure when/where this error would pop up, but may also be useful to include the original build requirement in the message.